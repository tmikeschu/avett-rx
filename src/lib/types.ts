import { User } from "api";

//eslint-disable-next-line
export type Any = any;

export interface AuthedUser extends User {
  token: string;
}
