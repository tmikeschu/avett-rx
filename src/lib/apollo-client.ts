import * as React from "react";
import { ApolloClient, HttpLink, InMemoryCache } from "@apollo/client";

import { Any } from "./types";

type Cache = Record<string, Any>;
type InitialState = Record<string, Any>;

let apolloClient: ApolloClient<Cache>;

function createApolloClient(): ApolloClient<Cache> {
  return new ApolloClient({
    ssrMode: typeof window === "undefined",
    link: new HttpLink({
      uri: "https://graphql.fauna.com/graphql", // Server URL (must be absolute)
      credentials: "same-origin", // Additional fetch() options like `credentials` or `headers`
      headers: {
        Authorization: `Bearer ${process.env.NEXT_PUBLIC_FAUNA_VISITOR_KEY}`,
      },
    }),
    cache: new InMemoryCache({}),
  });
}

export function initializeApollo(
  initialState: InitialState = null
): ApolloClient<Cache> {
  const _apolloClient = apolloClient ?? createApolloClient();

  // If your page has Next.js data fetching methods that use Apollo Client, the initial state
  // gets hydrated here
  if (initialState) {
    // Get existing cache, loaded during client side data fetching
    const existingCache = _apolloClient.extract();
    // Restore the cache using the data passed from getStaticProps/getServerSideProps
    // combined with the existing cached data
    _apolloClient.cache.restore({ ...existingCache, ...initialState });
  }
  // For SSG and SSR always create a new Apollo Client
  if (typeof window === "undefined") return _apolloClient;
  // Create the Apollo Client once in the client
  if (!apolloClient) apolloClient = _apolloClient;

  return _apolloClient;
}

export function useApollo(initialState: InitialState): ApolloClient<Cache> {
  const store = React.useMemo(() => initializeApollo(initialState), [
    initialState,
  ]);
  return store;
}
