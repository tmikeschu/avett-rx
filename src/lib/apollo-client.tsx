import * as React from "react";
import {
  ApolloClient,
  ApolloProvider as AP,
  HttpLink,
  InMemoryCache,
} from "@apollo/client";

import { Any } from "./types";
import useCurrentUser from "./use-current-user";

type Cache = Record<string, Any>;
export type InitialState = Record<string, Any>;

let apolloClient: ApolloClient<Cache>;

function createApolloClient({
  token,
}: {
  token?: string;
} = {}): ApolloClient<Cache> {
  return new ApolloClient({
    ssrMode: typeof window === "undefined",
    link: new HttpLink({
      uri: process.env.NEXT_PUBLIC_FAUNA_GRAPHQL_URI,
      credentials: "same-origin",
      headers: {
        Authorization: `Bearer ${
          token || process.env.NEXT_PUBLIC_FAUNA_VISITOR_KEY
        }`,
      },
      ...(process.env.NODE_ENV === "test"
        ? { fetch: require("cross-fetch") }
        : {}),
    }),
    cache: new InMemoryCache({}),
  });
}

export function initializeApollo({
  token,
  ...initialState
}: InitialState = {}): ApolloClient<Cache> {
  const _apolloClient = apolloClient ?? createApolloClient({ token });

  // If your page has Next.js data fetching methods that use Apollo Client, the initial state
  // gets hydrated here
  if (initialState) {
    // Get existing cache, loaded during client side data fetching
    const existingCache = _apolloClient.extract();
    // Restore the cache using the data passed from getStaticProps/getServerSideProps
    // combined with the existing cached data
    _apolloClient.cache.restore({ ...existingCache, ...initialState });
  }
  // For SSG and SSR always create a new Apollo Client
  if (typeof window === "undefined") return _apolloClient;
  // Create the Apollo Client once in the client
  if (!apolloClient) apolloClient = _apolloClient;

  return _apolloClient;
}

export function useApollo(initialState: InitialState): ApolloClient<Cache> {
  const { user = { token: "" } } = useCurrentUser();

  const store = React.useMemo(
    () => initializeApollo({ ...initialState, token: user.token }),
    [initialState, user.token]
  );

  return store;
}

export const ApolloProvider: React.FC<{
  initialState?: InitialState;
}> = ({ children, initialState = {} }) => {
  const client = useApollo(initialState);
  return <AP client={client}>{children}</AP>;
};
