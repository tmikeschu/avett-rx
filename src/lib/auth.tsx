import * as React from "react";
import { Magic } from "magic-sdk";
import Router from "next/router";

import { Any } from "./types";
import useCurrentUser from "./use-current-user";
import { createUsableContext } from "./utils";

const M =
  typeof window === "undefined"
    ? undefined
    : new Magic(process.env.NEXT_PUBLIC_MAGIC_PUBLISHABLE_KEY || "");

export type AuthContext = {
  user?: Any;
  status: "loading" | "ready";
  login: (email: string) => Promise<void>;
  logout: () => Promise<void>;
  client?: Magic;
};
export const [useAuth, AuthContext] = createUsableContext<AuthContext>();

const logout = async () => {
  fetch("/api/logout");
};

const login = async (email: string): Promise<void> => {
  const body = { email };

  try {
    const didToken = await M?.auth.loginWithMagicLink({
      email: body.email,
    });
    const res = await fetch("/api/login", {
      method: "POST",
      headers: {
        "Content-Type": "application/json",
        Authorization: "Bearer " + didToken,
      },
      body: JSON.stringify(body),
    });
    if (res.status === 200) {
      return res.json();
    } else {
      throw new Error(await res.text());
    }
  } catch (error) {
    console.error("An unexpected error happened occurred:", error);
  }
};

export const AuthProvider: React.FC = ({ children }) => {
  const [state, setState] = React.useState<AuthContext>({
    status: "loading",
    user: undefined,
    client: M,
    login: async (email: string) => {
      const user = await login(email);
      setState((s) => ({ ...s, user }));
    },
    logout: async () => {
      await logout();
      setState((s) => ({ ...s, user: undefined }));
    },
  });
  const currentUser = useCurrentUser();

  React.useEffect(() => {
    setState((s) => ({ ...s, status: "ready" }));
  }, []);

  React.useEffect(() => {
    setState((s) => ({ ...s, user: currentUser }));
  }, [currentUser]);

  return <AuthContext.Provider value={state}>{children}</AuthContext.Provider>;
};

export const MockAuthProvider: React.FC<{
  initialState?: Partial<AuthContext>;
}> = ({ children, initialState }) => {
  const mockLogin = React.useCallback(async (): Promise<void> => {
    setState((s) => ({ ...s, user: {} }));
  }, []);

  const mockLogout = React.useCallback(async (): Promise<void> => {
    setState((s) => ({ ...s, user: undefined }));
  }, []);

  const [state, setState] = React.useState<AuthContext>({
    status: "ready",
    user: undefined,
    client: undefined,
    login: mockLogin,
    logout: mockLogout,
    ...initialState,
  });

  return <AuthContext.Provider value={state}>{children}</AuthContext.Provider>;
};
