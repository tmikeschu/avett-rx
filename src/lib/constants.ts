export const COLORS = [
  "dark",
  "light",
  "primary",
  "secondary",
  "cancel",
  "warning",
  "error",
  "success",
] as const;
export type Color = typeof COLORS[number];
