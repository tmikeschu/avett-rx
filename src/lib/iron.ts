import Iron from "@hapi/iron";
import { NextApiRequest } from "next";

import { getTokenCookie } from "./auth-cookies";
import { Any } from "./types";

// Use an environment variable here instead of a hardcoded value for production
const TOKEN_SECRET = "this-is-a-secret-value-with-at-least-32-characters";

export function encryptSession(session: Record<string, Any>): Promise<string> {
  return Iron.seal(session, TOKEN_SECRET, Iron.defaults);
}

export async function getSession(
  req: NextApiRequest
): Promise<Record<string, Any>> {
  const token = getTokenCookie(req);
  return token && Iron.unseal(token, TOKEN_SECRET, Iron.defaults);
}
