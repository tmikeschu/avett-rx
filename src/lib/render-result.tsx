import * as React from "react";
import { ApolloError, QueryResult } from "@apollo/client";

const isEmpty = (data: NonNullable<QueryResult["data"]>): boolean => {
  const dataValue = data[Object.keys(data)[0]];
  return (
    dataValue === null || (Array.isArray(dataValue) && dataValue.length === 0)
  );
};

const successSelector = <T extends NonNullable<QueryResult["data"]>>(
  data: T
): T => data;

export const renderResult = <T extends QueryResult, U>(
  result: T,
  options: {
    isEmpty?: (data: NonNullable<T["data"]>) => boolean;
    successSelector?: (data: NonNullable<T["data"]>) => U;
    Loading: React.FC;
    Failure: React.FC<{ error: ApolloError }>;
    Empty: React.FC;
    Success: React.FC<{ data: U }>;
  }
): React.ReactElement => {
  return result.loading ? (
    <options.Loading />
  ) : result.error ? (
    <options.Failure error={result.error} />
  ) : (options.isEmpty ?? isEmpty)(result.data) ? (
    <options.Empty />
  ) : (
    <options.Success
      data={(options.successSelector ?? successSelector)(result.data)}
    />
  );
};
