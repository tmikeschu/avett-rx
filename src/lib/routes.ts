export enum Route {
  Root = "/",
  Pharmacy = "/pharmacy",
  ApiLogin = "/api/login",
  ApiLogout = "/api/logout",
  ApiUser = "/api/user",
  ApiGraphql = "/api/graphql",
}

export const VISITOR_VIEWS = {
  [Route.Root]: Route.Root,
  [Route.Pharmacy]: Route.Pharmacy,
} as const;

export const isVisitorView = (view: string): boolean => view in VISITOR_VIEWS;
