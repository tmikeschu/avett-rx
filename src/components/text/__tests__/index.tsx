import * as React from "react";

import { COLORS } from "lib/constants";
import * as utils from "test";

import Text, { VARIANTS } from "../";

describe("<Text />", () => {
  it("matches snapshot", () => {
    const { getByText } = utils.render(<Text>Snapshot</Text>);
    const p = getByText(/snapshot/i);
    expect(p).toMatchInlineSnapshot(`
      <p
        class="text-dark leading-normal font-body"
      >
        Snapshot
      </p>
    `);
  });

  it("can be rendered as a different component", () => {
    const { getByText } = utils.render(
      <Text variant="h1" as="span">
        Snapshot
      </Text>
    );
    const span = getByText(/snapshot/i);
    expect(span).toMatchInlineSnapshot(`
      <span
        class="text-dark text-4xl tracking-tighter font-display leading-tight"
      >
        Snapshot
      </span>
    `);
  });

  describe("variants and colors", () => {
    VARIANTS.forEach((variant) => {
      COLORS.forEach((color) => {
        it(`variant: ${variant}, color: ${color} matches snapshot`, () => {
          const { getByText } = utils.render(
            <Text variant={variant} color={color}>
              snapshot
            </Text>
          );
          const p = getByText(/snapshot/i);
          expect(p.className).toMatchSnapshot();
        });
      });
    });
  });
});
