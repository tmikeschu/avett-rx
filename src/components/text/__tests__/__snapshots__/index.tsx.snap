// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Text /> variants and colors variant: body1, color: cancel matches snapshot 1`] = `"text-cancel leading-normal font-body"`;

exports[`<Text /> variants and colors variant: body1, color: dark matches snapshot 1`] = `"text-dark leading-normal font-body"`;

exports[`<Text /> variants and colors variant: body1, color: error matches snapshot 1`] = `"text-error leading-normal font-body"`;

exports[`<Text /> variants and colors variant: body1, color: light matches snapshot 1`] = `"text-light leading-normal font-body"`;

exports[`<Text /> variants and colors variant: body1, color: primary matches snapshot 1`] = `"text-primary leading-normal font-body"`;

exports[`<Text /> variants and colors variant: body1, color: secondary matches snapshot 1`] = `"text-secondary leading-normal font-body"`;

exports[`<Text /> variants and colors variant: body1, color: success matches snapshot 1`] = `"text-success leading-normal font-body"`;

exports[`<Text /> variants and colors variant: body1, color: warning matches snapshot 1`] = `"text-dark border-warning border-b-4 border-solid leading-normal font-body"`;

exports[`<Text /> variants and colors variant: button, color: cancel matches snapshot 1`] = `"text-cancel font-bold font-body uppercase"`;

exports[`<Text /> variants and colors variant: button, color: dark matches snapshot 1`] = `"text-dark font-bold font-body uppercase"`;

exports[`<Text /> variants and colors variant: button, color: error matches snapshot 1`] = `"text-error font-bold font-body uppercase"`;

exports[`<Text /> variants and colors variant: button, color: light matches snapshot 1`] = `"text-light font-bold font-body uppercase"`;

exports[`<Text /> variants and colors variant: button, color: primary matches snapshot 1`] = `"text-primary font-bold font-body uppercase"`;

exports[`<Text /> variants and colors variant: button, color: secondary matches snapshot 1`] = `"text-secondary font-bold font-body uppercase"`;

exports[`<Text /> variants and colors variant: button, color: success matches snapshot 1`] = `"text-success font-bold font-body uppercase"`;

exports[`<Text /> variants and colors variant: button, color: warning matches snapshot 1`] = `"text-dark border-warning border-b-4 border-solid font-bold font-body uppercase"`;

exports[`<Text /> variants and colors variant: caption, color: cancel matches snapshot 1`] = `"text-cancel text-xs text-gray-600 font-body"`;

exports[`<Text /> variants and colors variant: caption, color: dark matches snapshot 1`] = `"text-dark text-xs text-gray-600 font-body"`;

exports[`<Text /> variants and colors variant: caption, color: error matches snapshot 1`] = `"text-error text-xs text-gray-600 font-body"`;

exports[`<Text /> variants and colors variant: caption, color: light matches snapshot 1`] = `"text-light text-xs text-gray-600 font-body"`;

exports[`<Text /> variants and colors variant: caption, color: primary matches snapshot 1`] = `"text-primary text-xs text-gray-600 font-body"`;

exports[`<Text /> variants and colors variant: caption, color: secondary matches snapshot 1`] = `"text-secondary text-xs text-gray-600 font-body"`;

exports[`<Text /> variants and colors variant: caption, color: success matches snapshot 1`] = `"text-success text-xs text-gray-600 font-body"`;

exports[`<Text /> variants and colors variant: caption, color: warning matches snapshot 1`] = `"text-dark border-warning border-b-4 border-solid text-xs text-gray-600 font-body"`;

exports[`<Text /> variants and colors variant: h1, color: cancel matches snapshot 1`] = `"text-cancel text-4xl tracking-tighter font-display leading-tight"`;

exports[`<Text /> variants and colors variant: h1, color: dark matches snapshot 1`] = `"text-dark text-4xl tracking-tighter font-display leading-tight"`;

exports[`<Text /> variants and colors variant: h1, color: error matches snapshot 1`] = `"text-error text-4xl tracking-tighter font-display leading-tight"`;

exports[`<Text /> variants and colors variant: h1, color: light matches snapshot 1`] = `"text-light text-4xl tracking-tighter font-display leading-tight"`;

exports[`<Text /> variants and colors variant: h1, color: primary matches snapshot 1`] = `"text-primary text-4xl tracking-tighter font-display leading-tight"`;

exports[`<Text /> variants and colors variant: h1, color: secondary matches snapshot 1`] = `"text-secondary text-4xl tracking-tighter font-display leading-tight"`;

exports[`<Text /> variants and colors variant: h1, color: success matches snapshot 1`] = `"text-success text-4xl tracking-tighter font-display leading-tight"`;

exports[`<Text /> variants and colors variant: h1, color: warning matches snapshot 1`] = `"text-dark border-warning border-b-4 border-solid text-4xl tracking-tighter font-display leading-tight"`;

exports[`<Text /> variants and colors variant: h2, color: cancel matches snapshot 1`] = `"text-cancel text-2xl tracking-tighter font-display leading-tight"`;

exports[`<Text /> variants and colors variant: h2, color: dark matches snapshot 1`] = `"text-dark text-2xl tracking-tighter font-display leading-tight"`;

exports[`<Text /> variants and colors variant: h2, color: error matches snapshot 1`] = `"text-error text-2xl tracking-tighter font-display leading-tight"`;

exports[`<Text /> variants and colors variant: h2, color: light matches snapshot 1`] = `"text-light text-2xl tracking-tighter font-display leading-tight"`;

exports[`<Text /> variants and colors variant: h2, color: primary matches snapshot 1`] = `"text-primary text-2xl tracking-tighter font-display leading-tight"`;

exports[`<Text /> variants and colors variant: h2, color: secondary matches snapshot 1`] = `"text-secondary text-2xl tracking-tighter font-display leading-tight"`;

exports[`<Text /> variants and colors variant: h2, color: success matches snapshot 1`] = `"text-success text-2xl tracking-tighter font-display leading-tight"`;

exports[`<Text /> variants and colors variant: h2, color: warning matches snapshot 1`] = `"text-dark border-warning border-b-4 border-solid text-2xl tracking-tighter font-display leading-tight"`;

exports[`<Text /> variants and colors variant: h3, color: cancel matches snapshot 1`] = `"text-cancel text-xl tracking-tighter font-display leading-tight"`;

exports[`<Text /> variants and colors variant: h3, color: dark matches snapshot 1`] = `"text-dark text-xl tracking-tighter font-display leading-tight"`;

exports[`<Text /> variants and colors variant: h3, color: error matches snapshot 1`] = `"text-error text-xl tracking-tighter font-display leading-tight"`;

exports[`<Text /> variants and colors variant: h3, color: light matches snapshot 1`] = `"text-light text-xl tracking-tighter font-display leading-tight"`;

exports[`<Text /> variants and colors variant: h3, color: primary matches snapshot 1`] = `"text-primary text-xl tracking-tighter font-display leading-tight"`;

exports[`<Text /> variants and colors variant: h3, color: secondary matches snapshot 1`] = `"text-secondary text-xl tracking-tighter font-display leading-tight"`;

exports[`<Text /> variants and colors variant: h3, color: success matches snapshot 1`] = `"text-success text-xl tracking-tighter font-display leading-tight"`;

exports[`<Text /> variants and colors variant: h3, color: warning matches snapshot 1`] = `"text-dark border-warning border-b-4 border-solid text-xl tracking-tighter font-display leading-tight"`;

exports[`<Text /> variants and colors variant: h4, color: cancel matches snapshot 1`] = `"text-cancel text-lg tracking-tight font-display leading-tight"`;

exports[`<Text /> variants and colors variant: h4, color: dark matches snapshot 1`] = `"text-dark text-lg tracking-tight font-display leading-tight"`;

exports[`<Text /> variants and colors variant: h4, color: error matches snapshot 1`] = `"text-error text-lg tracking-tight font-display leading-tight"`;

exports[`<Text /> variants and colors variant: h4, color: light matches snapshot 1`] = `"text-light text-lg tracking-tight font-display leading-tight"`;

exports[`<Text /> variants and colors variant: h4, color: primary matches snapshot 1`] = `"text-primary text-lg tracking-tight font-display leading-tight"`;

exports[`<Text /> variants and colors variant: h4, color: secondary matches snapshot 1`] = `"text-secondary text-lg tracking-tight font-display leading-tight"`;

exports[`<Text /> variants and colors variant: h4, color: success matches snapshot 1`] = `"text-success text-lg tracking-tight font-display leading-tight"`;

exports[`<Text /> variants and colors variant: h4, color: warning matches snapshot 1`] = `"text-dark border-warning border-b-4 border-solid text-lg tracking-tight font-display leading-tight"`;

exports[`<Text /> variants and colors variant: h5, color: cancel matches snapshot 1`] = `"text-cancel text-sm tracking-tight font-display leading-tight"`;

exports[`<Text /> variants and colors variant: h5, color: dark matches snapshot 1`] = `"text-dark text-sm tracking-tight font-display leading-tight"`;

exports[`<Text /> variants and colors variant: h5, color: error matches snapshot 1`] = `"text-error text-sm tracking-tight font-display leading-tight"`;

exports[`<Text /> variants and colors variant: h5, color: light matches snapshot 1`] = `"text-light text-sm tracking-tight font-display leading-tight"`;

exports[`<Text /> variants and colors variant: h5, color: primary matches snapshot 1`] = `"text-primary text-sm tracking-tight font-display leading-tight"`;

exports[`<Text /> variants and colors variant: h5, color: secondary matches snapshot 1`] = `"text-secondary text-sm tracking-tight font-display leading-tight"`;

exports[`<Text /> variants and colors variant: h5, color: success matches snapshot 1`] = `"text-success text-sm tracking-tight font-display leading-tight"`;

exports[`<Text /> variants and colors variant: h5, color: warning matches snapshot 1`] = `"text-dark border-warning border-b-4 border-solid text-sm tracking-tight font-display leading-tight"`;

exports[`<Text /> variants and colors variant: h6, color: cancel matches snapshot 1`] = `"text-cancel text-xs font-display leading-tight"`;

exports[`<Text /> variants and colors variant: h6, color: dark matches snapshot 1`] = `"text-dark text-xs font-display leading-tight"`;

exports[`<Text /> variants and colors variant: h6, color: error matches snapshot 1`] = `"text-error text-xs font-display leading-tight"`;

exports[`<Text /> variants and colors variant: h6, color: light matches snapshot 1`] = `"text-light text-xs font-display leading-tight"`;

exports[`<Text /> variants and colors variant: h6, color: primary matches snapshot 1`] = `"text-primary text-xs font-display leading-tight"`;

exports[`<Text /> variants and colors variant: h6, color: secondary matches snapshot 1`] = `"text-secondary text-xs font-display leading-tight"`;

exports[`<Text /> variants and colors variant: h6, color: success matches snapshot 1`] = `"text-success text-xs font-display leading-tight"`;

exports[`<Text /> variants and colors variant: h6, color: warning matches snapshot 1`] = `"text-dark border-warning border-b-4 border-solid text-xs font-display leading-tight"`;

exports[`<Text /> variants and colors variant: overline, color: cancel matches snapshot 1`] = `"text-cancel uppercase text-sm font-body"`;

exports[`<Text /> variants and colors variant: overline, color: dark matches snapshot 1`] = `"text-dark uppercase text-sm font-body"`;

exports[`<Text /> variants and colors variant: overline, color: error matches snapshot 1`] = `"text-error uppercase text-sm font-body"`;

exports[`<Text /> variants and colors variant: overline, color: light matches snapshot 1`] = `"text-light uppercase text-sm font-body"`;

exports[`<Text /> variants and colors variant: overline, color: primary matches snapshot 1`] = `"text-primary uppercase text-sm font-body"`;

exports[`<Text /> variants and colors variant: overline, color: secondary matches snapshot 1`] = `"text-secondary uppercase text-sm font-body"`;

exports[`<Text /> variants and colors variant: overline, color: success matches snapshot 1`] = `"text-success uppercase text-sm font-body"`;

exports[`<Text /> variants and colors variant: overline, color: warning matches snapshot 1`] = `"text-dark border-warning border-b-4 border-solid uppercase text-sm font-body"`;

exports[`<Text /> variants and colors variant: subtitle, color: cancel matches snapshot 1`] = `"text-cancel text-sm font-body"`;

exports[`<Text /> variants and colors variant: subtitle, color: dark matches snapshot 1`] = `"text-dark text-sm font-body"`;

exports[`<Text /> variants and colors variant: subtitle, color: error matches snapshot 1`] = `"text-error text-sm font-body"`;

exports[`<Text /> variants and colors variant: subtitle, color: light matches snapshot 1`] = `"text-light text-sm font-body"`;

exports[`<Text /> variants and colors variant: subtitle, color: primary matches snapshot 1`] = `"text-primary text-sm font-body"`;

exports[`<Text /> variants and colors variant: subtitle, color: secondary matches snapshot 1`] = `"text-secondary text-sm font-body"`;

exports[`<Text /> variants and colors variant: subtitle, color: success matches snapshot 1`] = `"text-success text-sm font-body"`;

exports[`<Text /> variants and colors variant: subtitle, color: warning matches snapshot 1`] = `"text-dark border-warning border-b-4 border-solid text-sm font-body"`;
